src/autopas/particles/ParticleBase.h

```diff
@@ -152,6 +152,23 @@ class ParticleBase {
    */
   void setR(const std::array<double, 3> &r) { _r = r; }
 
+  /**
+   * Add a distance vector to the position of the particle and check if the distance between the old and new position
+   * is less than a given max distance.
+   * This max distance usually should be the skin per timestep divided by two.
+   *
+   * @param r vector to be added
+   * @param maxDistSquared The maximum allowed movement distance squared.
+   * @return true if dot(r - _r) < skinPerTimestepHalvedSquared
+   */
+  bool setRDistanceOk(const std::array<double, 3> &r, double maxDistSquared) {
```

Comment:
Not keen on the naming. Some alternative suggestions:

- `setRWithCheck`
- `setRWithDistanceCheck`
- `setRSafe` (except it is only "safe" if the user does something with this information)

Reply:
I'm also not really happy with the name but I noticed that something like `setRWithCheck` doesn't nicely convey the whats happening when used in an `if`. Maybe `setRSafe` is a good middleground?


```c++
if (setRWithCheck(r)) { /* do stuff */ }
// vs
if (setRDistanceOk(r)) { /* do stuff */ }
// vs
if (setRSafe(r)) { /* do stuff */ }
```

Reply:
My concern with `setRSafe` is that it implies that the set *is* safe when in reality it just warns if the set if unsafe. I would prefer `setRDistanceCheck` as a compromise.

Reply:
renamed to `setRDistanceCheck` fd40e01